x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/add_domain/add_domain_logic.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { i18n } from '@kbn/i18n';
+
+import { flashSuccessToast } from '../../../../../shared/flash_messages';
+import { defaultErrorMessage } from '../../../../../shared/flash_messages/handle_api_errors';
+
+import { HttpLogic } from '../../../../../shared/http';
+import { KibanaLogic } from '../../../../../shared/kibana';
+import { ENGINE_CRAWLER_DOMAIN_PATH } from '../../../../routes';
+import { EngineLogic, generateEnginePath } from '../../../engine';
+
+import { CrawlerOverviewLogic } from '../../crawler_overview_logic';
+import { CrawlerDataFromServer, CrawlerDomain } from '../../types';
+import { crawlerDataServerToClient } from '../../utils';
+
+import { extractDomainAndEntryPointFromUrl } from './utils';
+
+export interface AddDomainLogicValues {
+  addDomainFormInputValue: string;
+  allowSubmit: boolean;
+  hasValidationCompleted: boolean;
+  entryPointValue: string;
+  errors: string[];
+}
+
+export interface AddDomainLogicActions {
+  clearDomainFormInputValue(): void;
+  setAddDomainFormInputValue(newValue: string): string;
+  onSubmitNewDomainError(errors: string[]): { errors: string[] };
+  onSubmitNewDomainSuccess(domain: CrawlerDomain): { domain: CrawlerDomain };
+  onValidateDomain(
+    newValue: string,
+    newEntryPointValue: string
+  ): { newValue: string; newEntryPointValue: string };
+  submitNewDomain(): void;
+  validateDomain(): void;
+}
+
+export const AddDomainLogic = kea<MakeLogicType<AddDomainLogicValues, AddDomainLogicActions>>({
+  path: ['enterprise_search', 'app_search', 'crawler', 'add_domain'],
+  actions: () => ({
+    clearDomainFormInputValue: true,
+    setAddDomainFormInputValue: (newValue) => newValue,
+    onSubmitNewDomainSuccess: (domain) => ({ domain }),
+    onSubmitNewDomainError: (errors) => ({ errors }),
+    onValidateDomain: (newValue, newEntryPointValue) => ({
+      newValue,
+      newEntryPointValue,
+    }),
+    submitNewDomain: true,
+    validateDomain: true,
+  }),
+  reducers: () => ({
+    addDomainFormInputValue: [
+      'https://',
```

Comment:
Totally optional: You could add `'https://'` and `/` in entryPointValue to a constant in this file so you're not repeating them.

Reply:
I went ahead and did this in 6d32f8c

Reply:
/shrug Haha, yeah, good point.